inductiva/structures/post_processing/fenicsx.py

```diff
@@ -0,0 +1,149 @@
+        values directly.
+        It reads an XDMF file, removes lines containing Time Values, and writes
+        the modified content to a new file.
+
+        Args: 
+            output_file_path: The path to save the modified XDMF content.
+        """
+
+        # Read the XDMF file
+        with open(self.sim_output_path, "r", encoding="utf-8") as file:
+            xdmf_content = file.readlines()
+
+        # Create an empty list to store the modified content
+        new_xdmf_content = []
+
+        # Iterate through the lines in the file
+        for line in xdmf_content:
+
+            # Check if the line contains a Time Value
+            if "<Time Value=" not in line:
+                new_xdmf_content.append(line)
+
+        # Write the modified content back to the output file
+        with open(output_file_path, "w", encoding="utf-8") as file:
+            file.writelines(new_xdmf_content)
+
+    @inductiva.utils.optional_deps.needs_structures_extra_deps
+    def render(self,
+               field_names: list = None,
+               scalar_bar: bool = True,
+               transparent_background: bool = True) -> None:
+        """Render the simulation fields as images.
+
+        Args:
+            field_names (list, optional): List of field names to render. Default
+              is None.
+            scalar_bar (bool, optional): Whether to include a scalar bar legend
+              in the visualization. Default is True.
+            transparent_background (bool, optional): Whether the background of
+              the saved images should be transparent. Default is True.
+        """
+
+        # Create a temporary directory
+        temp_dir = tempfile.mkdtemp()
+
+        # Specify the filename within the temporary directory
+        temp_file_path = os.path.join(temp_dir, "temp_results.xdmf")
+
+        # Remove time values from the XDMF file
+        self.remove_time_values_from_xdmf(self.sim_output_path)
+
+        # Read simulation output data
+        reader = pv.get_reader(temp_file_path)
+        data = reader.read()
+
+        # Clean up temporary files and directory
+        os.remove(temp_file_path)
+        os.rmdir(temp_dir)
+
+        if field_names is None:
+
+            # Default field names to render
+            field_names = [
+                "displacement_x", "displacement_y", "displacement_z",
+                "stress_xx", "stress_xy", "stress_xz", "stress_yx", "stress_yy",
+                "stress_yz", "stress_zx", "stress_zy", "stress_zz", "strain_xx",
+                "strain_xy", "strain_xz", "strain_yx", "strain_yy", "strain_yz",
+                "strain_zx", "strain_zy", "strain_zz", "von_mises"
+            ]
+
+        # Get the list of available keys (field names) in the PyVista dataset
+        keys = list(data.keys())
+
+        # Iterate through the specified field names to render
+        for field_name in field_names:
```

Comment:
If I understand this, it will just create 21 graphs when field_names is None. Don't you think this is super cumbersome for the user (you)?

Reply:
I understand your point, the idea here was to provide all the outputs as pictures in case people don't want to specify what they want to see. If you know what you want to see, you only need to specify the name. Do you think there's a better alternative?

Reply:
Yes, but with pyvista a little widget appears, so I need to just down 21 images before I move with my life. That is more painful, that selecting what I want to see. If you say to the user that he can see the stress, displacements, etc and the possible planes, then he can configure it himself. If you want to retrieve a list, that consider passing these args as a list.